ionic2/app/pages/wiki-index/wiki-index.page.ts

```diff
@@ -0,0 +1,38 @@
+import { forwardRef } from '@angular/core';
+import { NavController, Page } from 'ionic-angular';
+import { Observable, Subscription } from 'rxjs';
+import { PokeDetailPage } from "../pokedetail/pokedetail.page";
+import { Pokemon } from "../../models/pokemon";
+import { NavbarComponent } from '../../components/navbar/navbar.component';
+import { ApiService } from '../../services/api.service';
+
+@Page({
+  templateUrl: 'pages/wiki-index/wiki-index.page.html',
+  directives: [forwardRef(() => NavbarComponent)]
+})
+export class WikiIndexPage {
+
+  queryString: string;
+  querySubscription: Subscription;
+  results: Pokemon[] = [];
+
+  constructor(private navCtrl: NavController, private apiservice: ApiService) { }
+
+  onInput($event) {
+    this.cancelRequests();
+      // Search Pokemon
+      this.querySubscription = this.apiservice.getPokemonByName(this.queryString)
+        .subscribe(results => this.results = results, error => this.results = []);
+  }
+
+  cancelRequests() {
+    if (this.querySubscription && !this.querySubscription.isUnsubscribed) {
+      this.querySubscription.unsubscribe();
+    }
+  }
+
+  selectPokemon(pokemon:Pokemon) {
+    this.cancelRequests();
+    this.navCtrl.setRoot(PokeDetailPage, {pokemonId: pokemon.pokemonId});
```

Comment:
I think we should pop the page here. This will give us a nice back button on the PokeDetailPage.